lib/filesystem/find_steam_game.h

```diff
@@ -0,0 +1,262 @@
+/*
+Copyright (c) 2021 Cong
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+*/
+#pragma once
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
+#include <stdbool.h>
+#include <stdio.h>
+#include <string.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+
+#if defined(_WIN64) || defined(_WIN32)
+#define _FSG_FUNC static __inline
+#elif !defined __STDC_VERSION__ || __STDC_VERSION__ < 199901L
+#define _FSG_FUNC static __inline__
+#else
+#define _FSG_FUNC static inline
+#endif
+
+#if defined(_WIN64) || defined(_WIN32)
+#define WIN32_LEAN_AND_MEAN
+#include <windows.h>
+#include <winerror.h>
+#include <winreg.h>
+
+	_FSG_FUNC
+	void _fsg_query_reg_key(
+		char *out, const HKEY key, const char *keypath, const char *valname)
+	{
+		HKEY pathkey;
+		DWORD pathtype;
+		DWORD pathlen;
+		LONG res;
+
+		if (ERROR_SUCCESS ==
+			RegOpenKeyEx(key, keypath, 0, KEY_QUERY_VALUE, &pathkey))
+		{
+			if (ERROR_SUCCESS ==
+					RegQueryValueEx(
+						pathkey, valname, 0, &pathtype, NULL, &pathlen) &&
+				pathtype == REG_SZ && pathlen != 0)
+			{
+				res = RegQueryValueEx(
+					pathkey, valname, 0, NULL, (LPBYTE)out, &pathlen);
+				if (res != ERROR_SUCCESS)
+				{
+					out[0] = '\0';
+				}
+			}
+			RegCloseKey(pathkey);
+		}
+	}
+#else
+
+#include <pwd.h>
+#include <unistd.h>
+
+#endif
+
+#if defined(_WIN64) || defined(_WIN32)
+#define _FSG_PATH_MAX MAX_PATH
+#elif defined __linux__
+#include <limits.h>
+#ifdef PATH_MAX
+#define _FSG_PATH_MAX PATH_MAX
+#endif
+#elif defined(__unix__) || (defined(__APPLE__) && defined(__MACH__))
+#include <sys/param.h>
+#if defined(BSD)
+#include <limits.h>
+#ifdef PATH_MAX
+#define _FSG_PATH_MAX PATH_MAX
+#endif
+#endif
+#endif
+
+#ifndef _FSG_PATH_MAX
+#define _FSG_PATH_MAX 4096
+#endif
+
+	_FSG_FUNC
+	bool _fsg_dir_exists(const char *dir)
+	{
+		struct stat info;
+		return stat(dir, &info) == 0 && (info.st_mode & S_IFDIR);
+	}
+
+	_FSG_FUNC
+	void fsg_get_steam_game_path(char *out, const char *name)
```

Comment:
If we do not use this method lets not include it. Another option is to use submodule

Reply:
We actually use submodules for dependencies not available as packages